flow/flow.cpp

```diff
@@ -21,9 +21,28 @@
 #include "flow/flow.h"
 #include "flow/DeterministicRandom.h"
 #include "flow/UnitTest.h"
+#include "flow/rte_memcpy.h"
+#include "flow/folly_memcpy.h"
 #include <stdarg.h>
 #include <cinttypes>
 
+#if (defined (__linux__) || defined (__FreeBSD__)) && defined(__AVX__)
+// For benchmarking; need a version of rte_memcpy that doesn't live in the same compilation unit as the test.
+void * rte_memcpy_noinline(void *__restrict __dest, const void *__restrict __src, size_t __n) {
+	return rte_memcpy(__dest, __src, __n);
+}
+
+// This compilation unit will be linked in to the main binary, so this should override glibc memcpy
+__attribute__((visibility ("default"))) void *memcpy (void *__restrict __dest, const void *__restrict __src, size_t __n) {
```

Comment:
Yes; the idea is to hijack memcpy everywhere.

In a future PR, I'd like to manually move small, dynamic length, hot memcpy calls to folly_memcpy or an inlined rte_memcpy.

I'm not sure which is better in practice.

For small copies where we don't know the length at compile time, folly_memcpy is faster than non-inlined rte_memcpy (the new default memcpy once this PR is merged).

However, it's not clear if inlined rte_memcpy beats folly_memcpy in such cases, since it pollutes the instruction cache.  (I won't be able to tell which works better in practice until I identify + modify some calls, but that's future work.)